src/DotVVM.Framework/Compilation/DefaultViewCompiler.cs

```diff
@@ -35,7 +35,7 @@ public DefaultViewCompiler(DotvvmConfiguration configuration)
         /// <summary>
         /// Compiles the view and returns a function that can be invoked repeatedly. The function builds full control tree and activates the page.
         /// </summary>
-        public virtual CSharpCompilation CompileView(string sourceCode, string fileName, CSharpCompilation compilation, string namespaceName, string className)
+        public virtual (ControlBuilderDescriptor, Func<CSharpCompilation>) CompileView(string sourceCode, string fileName, CSharpCompilation compilation, string namespaceName, string className)
```

Comment:
If you'd ever need additional properties, use the ControlBuilderDescriptor or attached property on CSharpCompilation. I don't see any reason to do that, but you still can, the function of the tuple is only to distinguish two steps of compilation - first phase, when tree is parsed and direstives read and second phase when the view is compiled. Only in case when you'd like to add additional state, it could potentially help you - but it wont, because the interface's consumer could not use that possibility anyway. And in case you'd also implement the consumer, you can add method to the interface or add another interface, as you would basically bypass this implementation.